code/game/objects/effects/effect_system/effects_sparks.dm

```diff
@@ -26,24 +26,50 @@
 	return INITIALIZE_HINT_LATELOAD
 
 /obj/effect/particle_effect/sparks/LateInitialize()
+	RegisterSignals(src, list(COMSIG_MOVABLE_CROSS, COMSIG_MOVABLE_CROSS_OVER), PROC_REF(sparks_touched))
 	flick(icon_state, src)
 	playsound(src, SFX_SPARKS, 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
 	var/turf/T = loc
 	if(isturf(T))
-		T.hotspot_expose(1000,100)
+		affect_location(T, TRUE) // just_initialized set to TRUE
 	QDEL_IN(src, 20)
 
 /obj/effect/particle_effect/sparks/Destroy()
 	var/turf/T = loc
 	if(isturf(T))
-		T.hotspot_expose(1000,100)
+		affect_location(T)
+	UnregisterSignal(src, list(COMSIG_MOVABLE_CROSS, COMSIG_MOVABLE_CROSS_OVER))
 	return ..()
 
 /obj/effect/particle_effect/sparks/Move()
 	..()
 	var/turf/T = loc
 	if(isturf(T))
-		T.hotspot_expose(1000,100)
+		affect_location(T)
+
+/obj/effect/particle_effect/sparks/proc/affect_location(turf/T, just_initialized = 0)
+	T.hotspot_expose(1000,100)
+	if(just_initialized)
+		for(var/atom/movable/singed in T)
+			sparks_touched(src, singed)
+
+
+/obj/effect/particle_effect/sparks/proc/sparks_touched(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+
+	if(isobj(AM))
```

Comment:
None of this should exist at all honestly it should all be proc overrides instead of typechecks

Reply:
I wanted proc overrides but there's nothing I can check against agnostically for both mob/living and obj/ vulnerability to flames